### PR TITLE
Create NAT rules and bridges

### PR DESCRIPTION
**Pull request by ysksuzuki**
This PR implements the following features

- Restructured directories for v2
- Create NAT rules
- Create network bridges specified with network specification

### CODE REVIEW

File: v2/Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.PHONY: code-check
+code-check: test-tools
+	test -z "$$(gofmt -s -l . | tee /dev/stderr)"
+	staticcheck ./...
+	test -z "$$(nilerr ./... 2>&1 | tee /dev/stderr)"
+	test -z "$$(custom-checker -restrictpkg.packages=html/template,log $$(go list -tags='$(GOTAGS)' ./...) 2>&1 | tee /dev/stderr)"
+	ineffassign .
```
Comment (ymmt2005) [inline v2/Makefile:9]: This can be replaced with `staticcheck`.

File: v2/Makefile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		cd /tmp; env GOFLAGS= GO111MODULE=on go get honnef.co/go/tools/cmd/staticcheck; \
+	fi
+
+.PHONY: nilerr
+nilerr:
+	if ! which nilerr >/dev/null; then \
+		cd /tmp; env GOFLAGS= GO111MODULE=on go get github.com/gostaticanalysis/nilerr/cmd/nilerr; \
+	fi
+
+.PHONY: ineffassign
```
Comment (ymmt2005) [inline v2/Makefile:44]: unnecessary

File: v2/pkg/dcnet/network.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+	"github.com/containernetworking/plugins/pkg/utils/sysctl"
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	maxNetworkNameLen = 15
+	v4ForwardKey      = "net.ipv4.ip_forward"
+	v6ForwardKey      = "net.ipv6.conf.all.forwarding"
```
Comment (ymmt2005) [inline v2/pkg/dcnet/network.go:15]: Use https://pkg.go.dev/github.com/containernetworking/plugins/pkg/ip#EnableIP4Forward

File: v2/pkg/dcnet/network.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+		if err != nil {
+			return nil, err
+		}
+		n.addr = addr
+	}
+
+	return n, nil
+}
+
+func validate(spec *NetworkSpec) error {
```
Comment (ymmt2005) [inline v2/pkg/dcnet/network.go:67]: make this a method of `NetworkSpec`.

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

**Comment by ysksuzuki** [comment]
Other review comments

- Remove vendors from v1
- Not need to check the current IP forward setting

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**